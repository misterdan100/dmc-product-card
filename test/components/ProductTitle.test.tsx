import React from "react";
import renderer from "react-test-renderer";
import {ProductTitle, ProductCard} from '../../src/components'
import { product1 } from "../data/products";


describe('ProductTitle', () => {
    test('show component with custom title', () => {
        const wrapper = renderer.create(
            <ProductTitle title="Custom title" />
        )

        expect(wrapper.toJSON()).toMatchSnapshot()
    })

    test('must showing the component with product name', () => {
        const wrapper = renderer.create(
            <ProductCard product={product1}>
                {
                    () => (
                        <ProductTitle />
                    )
                }
            </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot()

    })
})