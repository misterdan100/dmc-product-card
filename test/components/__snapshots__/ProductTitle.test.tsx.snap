// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle must showing the component with product name 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - without image
  </span>
</div>
`;

exports[`ProductTitle show component with custom title 1`] = `
<span
  className="productDescription undefined"
>
  Custom title
</span>
`;
